content/community/news/science-updates/2021-05-19-wagner-blog-post.md

```diff
@@ -0,0 +1,228 @@
+In this way, the primary owner is incentivized to provide thoughtful feedback on PRs.
+We’ve also become slightly better at the code review process as an organization now that most team members have been on both sides of the “Approve” button.
+But the big lesson-learned is that establishing a culture of reviews is as much a social and organizational challenge as a technical one.
+
+
+Two problems around code review that we’re encountering now (which don’t seem to be specific to OpenFF) are 1) the timeliness of PR reviews and 2) reserving personnel-time for PR reviews.
+Every active project on GitHub has a backlog of issues and PRs, and I don’t know at which point the number of open issues and PRs increases from “healthy project” to “we need to change how we’re doing this”.
+As of the time of this writing, I think our current Issue and PR numbers seem to be creeping towards the latter. 
+
+
+## Testing
+
+
+Another vector of effort that turns out to have a lot of overlap with code review culture is testing.
+
+
+My workflow for reviewing PRs has gradually evolved, and now the first thing I do is to check the file diff to read over the [tests that were changed or added](https://github.com/openforcefield/openff-toolkit/pull/904/files#diff-97b3d6d5bcbac9c7503e6851bb19d2167984ee382450fcf91bd156f9960b7aadR323-R338).
+Then I stare out a window for a minute or two and think about the implications of the new behavior, and whether it’s clearly the right thing without any further context.
+If it passes that check, then I take the mindset that the PR is basically approved unless there’s something really unexpected in the description or proposed implementation.
+
+
+But if the new behavior isn’t obvious and there’s a lot of history on the problem, or the PR changes an implementation but the change can’t be reflected in tests, then that’s when things get difficult.
+I’ll write about this below. 
+
+
+Testing has been a great help to us.
+The [MolSSI cookiecutter](https://github.com/MolSSI/cookiecutter-cms) deserves a ton of credit for providing clear instructions to scientific developers on how to set up and automate tests.
+Of course, there is some maintenance burden associated with the tests -- If we change a behavior, several tests might begin failing, and then we have to spend time determining why the tests are failing, and whether the failures are appropriate given our original intentions, or whether they indicate a deeper problem.
+In the worst cases, test failures caused by distant code changes reveal design flaws or undocumented assumptions in our models. 
+
+
+There are still many places where our testing can be expanded, but over the last year, the development of the test suite has helped me sleep much better at night.
+Our current code coverage is 88%, which means that 88% of our lines of source code are executed by at least one test.
```

Comment:
```suggestion
The current code coverage of the OpenFF Toolkit 88%, which means that 88% of our lines of source code are executed by at least one test.
```

One small clarification, since I have no idea what the coverage for **all** of our code is and that's how this read at first

Reply:
Good call. Thanks!